### ISSUE TITLE
Roslyn build failed with CS0234: 'AssemblyMetadataAttributeAttribute' does not exist in the namespace 'System.Reflection'

### ISSUE THREAD

**Issue by diryboy** [dotnet/roslyn#45927]
**Version Used**: 
VS2019 Version 16.7.0 Preview 3.1

```
> dotnet --list-sdks
1.0.3 [C:\Program Files\dotnet\sdk]
1.0.4 [C:\Program Files\dotnet\sdk]
2.1.802 [C:\Program Files\dotnet\sdk]
3.1.100 [C:\Program Files\dotnet\sdk]
3.1.200 [C:\Program Files\dotnet\sdk]
3.1.301 [C:\Program Files\dotnet\sdk]
3.1.400-preview-015178 [C:\Program Files\dotnet\sdk]
5.0.100-preview.6.20318.15 [C:\Program Files\dotnet\sdk]
```

**Steps to Reproduce**:

1.  Pull latest master, at the time c0b0187
2.  Restore.cmd
3.  Build.cmd

**Expected Behavior**:
Build success.

**Actual Behavior**:
```
Build FAILED.

H:\Projects\roslyn\artifacts\obj\ResultProvider.NetFX20\Debug\net20\ResultProvider.NetFX20.AssemblyInfo.cs(22,30): erro
r CS0234: The type or namespace name 'AssemblyMetadataAttributeAttribute' does not exist in the namespace 'System.Refle
ction' (are you missing an assembly reference?) [H:\Projects\roslyn\src\ExpressionEvaluator\Core\Source\ResultProvider\
NetFX20\ResultProvider.NetFX20.csproj]
H:\Projects\roslyn\artifacts\obj\ResultProvider.NetFX20\Debug\net20\ResultProvider.NetFX20.AssemblyInfo.cs(22,30): erro
r CS0234: The type or namespace name 'AssemblyMetadataAttribute' does not exist in the namespace 'System.Reflection' (a
re you missing an assembly reference?) [H:\Projects\roslyn\src\ExpressionEvaluator\Core\Source\ResultProvider\NetFX20\R
esultProvider.NetFX20.csproj]
```

**Additional Info**:
Contents of `roslyn\artifacts\obj\ResultProvider.NetFX20\Debug\net20\ResultProvider.NetFX20.AssemblyInfo.cs`:
```C#
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

[assembly: System.Reflection.AssemblyCompanyAttribute("Microsoft Corporation")]
[assembly: System.Reflection.AssemblyConfigurationAttribute("Debug")]
[assembly: System.Reflection.AssemblyCopyrightAttribute("© Microsoft Corporation. All rights reserved.")]
[assembly: System.Reflection.AssemblyFileVersionAttribute("42.42.42.42424")]
[assembly: System.Reflection.AssemblyInformationalVersionAttribute("3.8.0-dev")]
[assembly: System.Reflection.AssemblyProductAttribute("Microsoft.CodeAnalysis.ExpressionEvaluator.ResultProvider")]
[assembly: System.Reflection.AssemblyTitleAttribute("Microsoft.CodeAnalysis.ExpressionEvaluator.ResultProvider")]
[assembly: System.Reflection.AssemblyVersionAttribute("42.42.42.42")]
[assembly: System.Reflection.AssemblyMetadataAttribute("RepositoryUrl", "file://H:\\Projects\\roslyn\\")]
[assembly: System.Resources.NeutralResourcesLanguageAttribute("en-US")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.ResultProvider, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator.ResultProvider, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]

// Generated by the MSBuild WriteCodeFragment class.

```

**Comment by diryboy** [comment]
When I use Preview version of VS2019 Developer Command, build fails. When I use VS2019 Developer Command, build succeed. Might be something wrong in the preview?

**Comment by RikkiGibson** [comment]
IIRC we don't typically recommend using the developer command prompt to build Roslyn. I think the PATH changes in there can confuse it. Roslyn knows how to get everything it needs to build, including the precisely needed dotnet SDK version, from a regular command prompt.

**Comment by diryboy** [comment]
The build failed when using normal command prompt, so I tried developer command prompt. Just sharing the differences here. Anyways, this issue is old as I noticed the project is going to .NET 5 SDK. I assume the problem will go away once I have new SDK. But I'm having trouble getting a successful restore due to possible government censorship of azure.

I'm closing this issue for now, if I'm able to reproduce it again, I'll reopen it.
